extmod/moducryptolib.c

```diff
@@ -234,14 +309,30 @@ STATIC mp_obj_t aes_process(size_t n_args, const mp_obj_t *args, bool encrypt) {
         }
     }
 
-    if (self->block_mode == UCRYPTOLIB_MODE_ECB) {
+    switch (self->block_mode) {
+    case UCRYPTOLIB_MODE_ECB: {
         uint8_t *in = in_bufinfo.buf, *out = out_buf_ptr;
         uint8_t *top = in + in_bufinfo.len;
         for (; in < top; in += 16, out += 16) {
             aes_process_ecb_impl(&self->ctx, in, out, encrypt);
         }
-    } else {
+        break;
+    }
+
+    case UCRYPTOLIB_MODE_CBC:
         aes_process_cbc_impl(&self->ctx, in_bufinfo.buf, out_buf_ptr, in_bufinfo.len, encrypt);
+        break;
+
+    #if MICROPY_PY_UCRYPTOLIB_CTR
+    case UCRYPTOLIB_MODE_CTR:
+        aes_process_ctr_impl(&self->ctx, in_bufinfo.buf, out_buf_ptr, in_bufinfo.len,
+            ctr_params_from_aes(self));
+    #endif
+        break;
+
+    default:
+        MP_UNREACHABLE();
```

Comment:
This pattern is not used elsewhere in the code.  Usually default switch blocks either have an assert for the remaining case that's valid, or are combined with an existing valid case.

For this particular piece of code I'd suggest to just leave it as a set of if-else statements, and simply add a 3rd in between the 2 existing, like:
```c
#if MICROPY_PY_UCRYPTOLIB_CTR
} else if (self->block_mode == UCRYPTOLIB_MODE_CTR) {
    aes_process_ctr_impl(...);
#endif 
```
Then `MP_UNREACHABLE()` is not needed at all.

Reply:
My intention when adding the `MP_UNREACHABLE()` is that since the code became more branched now, and with conditional compilation as well, and it is possible that more branches will be added - Then the `MP_UNREACHABLE()` helps us identify misuses of the `switch` case more quickly (Compared to an `aes_process` call that just fails and return an empty bytes string, like what would happen now)

Reply:
I'll remove the `MP_UNREACHABLE` commit,

But I still rather keep it as a `switch`, looks much better IMO :)